aten/src/ATen/native/quantized/affine_quantizer_base.cpp

```diff
@@ -118,10 +118,10 @@ uint8_t quantize_val_arm(
     const float scale,
     const int32_t zero_point,
     const float value) {
-  const int32_t qmin = std::numeric_limits<uint8_t>::min();
-  const int32_t qmax = std::numeric_limits<uint8_t>::max();
+  constexpr int32_t qmin = std::numeric_limits<uint8_t>::min();
+  constexpr int32_t qmax = std::numeric_limits<uint8_t>::max();
   float inv_scale = 1.0f / scale;
-  auto r = zero_point + static_cast<int32_t>(Round(value * inv_scale));
+  auto r = static_cast<int32_t>(zero_point + Round(value * inv_scale));
```

Comment:
Yeah this was one of the changes we had to do but had fallen off of the radar. But it wont be sufficient to do it here. I would imagine that you need fix other places here: https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp. Because the vectorized arm code there does the same thing. Round and cast to int32 first and then add zero point. But doing there may have perf implication. That might be the reason why it was not done in the first place. Dont remember now. But landing just this change will change semantics between scalar and vectorized versions of quantize_val.

Reply:
Can you be a bit more specific what codepaths you are referring to?

I'm not 100% sure why overflow does not happen when `quantize_tensor_arm` perhaps because of the rounding trick described below:
https://github.com/pytorch/pytorch/blob/ddb1f293b688f5769b98f09c92eca7ddcab749e4/aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp#L2484

And vectorized part seems correct for all tests defined in https://github.com/pytorch/pytorch/blob/d62c3ea35447b16301dfab2383d6d09e8870bf86/test/quantization/core/test_quantized_tensor.py#L656